python/llm/example/CPU/LangChain/README.md

```diff
@@ -1,90 +1,155 @@
-## Langchain Examples
+# Langchain examples
 
-This folder contains examples showcasing how to use `langchain` with `ipex-llm`. 
+The examples in this folder shows how to use [LangChain](https://www.langchain.com/) with `ipex-llm` on Intel CPU.
 
-### Install-IPEX LLM
+## Streaming Chat Example
 
-Ensure `ipex-llm` is installed by following the [IPEX-LLM Installation Guide](https://ipex-llm.readthedocs.io/en/latest/doc/LLM/Overview/install_cpu.html). 
+### 1. Install Prerequisites
 
-### Install Dependences Required by the Examples
+To benefit from IPEX-LLM on Intel CPUs, there are several prerequisite steps for tools installation and environment preparation.
 
+Follow the instructions in [CPU Install Guide](../../../../../docs/mddocs/Overview/install_cpu.md) to install ipex-llm.
+
+### 2. Setting up Dependencies
+
+Install langchain dependencies:
+
+```bash
+pip install -U langchain langchain-community
+```
+
+### 3. Running the Streaming Chat Example
+
+In the current directory, run the example with command:
 
 ```bash
-pip install langchain==0.0.184
-pip install -U chromadb==0.3.25
-pip install -U pandas==2.0.3
+python chat.py -m MODEL_PATH -q QUESTION
 ```
+**Additional Parameters for Configuration:**
+- `-m MODEL_PATH`: **required**, path to the model
+- `-q QUESTION`: question to ask. Default is `What is AI?`.
+
+## Retrival Augmented Generation (RAG) Example
 
+### 1. Install Prerequisites
 
-### Example: Chat
+To benefit from IPEX-LLM on Intel CPUs, there are several prerequisite steps for tools installation and environment preparation.
 
-The chat example ([chat.py](./chat.py)) shows how to use `LLMChain` to build a chat pipeline. 
+Follow the instructions in [CPU Install Guide](../../../../../docs/mddocs/Overview/install_cpu.md) to install ipex-llm.
 
-To run the example, execute the following command in the current directory:
+### 2. Setting up Dependencies
+
+Install langchain dependencies:
 
 ```bash
-python chat.py -m <path_to_model> [-q <your_question>]
+pip install -U langchain langchain-community langchain-chroma sentence-transformers==3.0.1
```

Comment:
Why we need specific version here for `sentence-transformers`?

Reply:
Because the newer version of sentence-transformers will cause error when running. This is the last working version of sentence-transformers I found :)